providers/google/google.go

```diff
@@ -124,61 +148,108 @@ func (g *GoogleCloud) subscribe(opts ps.HandlerOptions, h ps.MsgHandler, ready c
 			Jitter: true,
 		}
 
+		type WorkerMsg struct {
+			ackID string
+			m     *pbpb.PubsubMessage
+			ctx   context.Context
+		}
+
 		// create a semaphore, this is because Google PubSub will spam
 		// your service if you can't process a message
 		// and will also not handle
 		sem := semaphore.NewWeighted(int64(opts.Concurrency))
+		pool := make(chan *WorkerMsg)
```

Comment:
Is `pool` the right name for this? This is just a 'queue' of messages waiting to be processed, right? `work`, `workQueue`, `messageQueue`? The handler runs in a separate goroutine, but we could also consider making this a buffered channel so that the subscription pull does not block (eg: if we pull messages faster than we consume them).